### PR TITLE
DBZ-7497 Document configuration based snapshot mode

### PR DESCRIPTION
**Pull request by mfvitale**
related to: https://github.com/debezium/debezium/pull/5422

### CODE REVIEW

File: documentation/modules/ROOT/pages/connectors/mongodb.adoc [hunk 1]
```diff
@@ -1993,13 +1998,47 @@ It does not transition to streaming event records for subsequent database change
...
+ifdef::community[]
+|[[mongodb-property-snapshot-mode-configuration-based-snapshot-on-schema-error]]<<mongodb-property-configuration-based-snapshot-on-schema-error, `+snapshot.mode.configuration.based.snapshot.on.schema.error+`>>
+|false
+|When `snapshot.mode` is set as `configuration_based`, this setting permits to specify whenever the schema should be snapshotted or not in case schema history topic is missing.
+endif::community[]
+
+ifdef::community[]
+|[[mongodb-property-snapshot-mode-configuration-based-snapshot-on-data-error]]<<mongodb-property-configuration-based-snapshot-on-data-error, `+snapshot.mode.configuration.based.snapshot.on.data.error+`>>
+|false
+|When `snapshot.mode` is set as `configuration_based`, this setting permits to specify whenever the data should be snapshotted or not in case data is no more present on db log.
```
[the remark this answers is not in the document]
Comment (roldanbob) [reply documentation/modules/ROOT/pages/connectors/mongodb.adoc:2039]: Same question as earlier.

File: documentation/modules/ROOT/pages/connectors/mysql.adoc [hunk 2]
```diff
@@ -697,6 +697,11 @@ WARNING: Do not use this mode to perform a snapshot if schema changes were commi
 * It cannot detect any topic offsets.
 * A previously recorded offset specifies a log position that is not available on the server.
 
+ifdef::community[]
+|`configuration_based`
+|The `configuration_based` snapshot mode allows control over snapshots by setting connectors properties prefixed with 'snapshot.mode.configuration.based'.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/mysql.adoc:702]:
```suggestion
|Set the snapshot mode to `configuration_based` to control snapshot behavior through the set of connector properties that have the prefix 'snapshot.mode.configuration.based'.
```

File: documentation/modules/ROOT/pages/connectors/sqlserver.adoc [hunk 3]
```diff
@@ -226,6 +226,11 @@ WARNING: Do not use this mode to perform a snapshot if schema changes were commi
 * It cannot detect any topic offsets.
 * A previously recorded offset specifies a log position that is not available on the server.
 
+ifdef::community[]
+|`configuration_based`
+|The `configuration_based` snapshot mode allows control over snapshots by setting connectors properties prefixed with 'snapshot.mode.configuration.based'.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/sqlserver.adoc:231]:
```suggestion
|Set the snapshot mode to `configuration_based` to control snapshot behavior through the set of connector properties that have the prefix 'snapshot.mode.configuration.based'.
```

File: documentation/modules/ROOT/pages/connectors/mysql.adoc [hunk 4]
```diff
@@ -3511,10 +3516,44 @@ This option is under consideration for future deprecation, in favor of the `no_d
...
+endif::community[]
+
 ifdef::community[]
 `custom`:: The connector performs a snapshot according to the implementation specified by the xref:mysql-property-snapshot-mode-custom-name[`snapshot.mode.custom.name`] property, which defines a custom implementation of the `io.debezium.spi.snapshot.Snapshotter` interface.
 endif::community[]
 
+ifdef::community[]
+|[[mysql-property-snapshot-mode-configuration-based-snapshot-data]]<<mysql-property-configuration-based-snapshot-data, `+snapshot.mode.configuration.based.snapshot.data+`>>
+|false
+|When `snapshot.mode` is set as `configuration_based`, this setting permits to specify whenever the data should be snapshotted or not.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/mysql.adoc:3530]:
```suggestion
|If the `snapshot.mode` is set to `configuration_based`, set this property to specify whether the connector includes table data when it performs a snapshot.
```

File: documentation/modules/ROOT/pages/connectors/mysql.adoc [hunk 5]
```diff
@@ -3511,10 +3516,44 @@ This option is under consideration for future deprecation, in favor of the `no_d
...
+endif::community[]
+
+ifdef::community[]
+|[[mysql-property-snapshot-mode-configuration-based-snapshot-schema]]<<mysql-property-configuration-based-snapshot-schema, `+snapshot.mode.configuration.based.snapshot.schema+`>>
+|false
+|When `snapshot.mode` is set as `configuration_based`, this setting permits to specify whenever the schema should be snapshotted or not.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/mysql.adoc:3536]:
```suggestion
|If the `snapshot.mode` is set to `configuration_based`, set this property to specify whether the connector includes the table schema when it performs a snapshot.
```

File: documentation/modules/ROOT/pages/connectors/mysql.adoc [hunk 6]
```diff
@@ -3511,10 +3516,44 @@ This option is under consideration for future deprecation, in favor of the `no_d
...
+endif::community[]
+
+ifdef::community[]
+|[[mysql-property-snapshot-mode-configuration-based-start-stream]]<<mysql-property-configuration-based-start-stream, `+snapshot.mode.configuration.based.start.stream+`>>
+|false
+|When `snapshot.mode` is set as `configuration_based`, this setting permits to specify whenever the stream should start or not after snapshot.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/mysql.adoc:3542]:
```suggestion
|If the `snapshot.mode` is set to `configuration_based`, set this property to specify whether the connector begins to stream change events after a snapshot completes.
```

File: documentation/modules/ROOT/pages/connectors/mysql.adoc [hunk 7]
```diff
@@ -3511,10 +3516,44 @@ This option is under consideration for future deprecation, in favor of the `no_d
...
+endif::community[]
+
+ifdef::community[]
+|[[mysql-property-snapshot-mode-configuration-based-snapshot-on-schema-error]]<<mysql-property-configuration-based-snapshot-on-schema-error, `+snapshot.mode.configuration.based.snapshot.on.schema.error+`>>
+|false
+|When `snapshot.mode` is set as `configuration_based`, this setting permits to specify whenever the schema should be snapshotted or not in case schema history topic is missing.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/mysql.adoc:3548]:
```suggestion
|If the `snapshot.mode` is set to `configuration_based`, set this property to specify whether the connector includes table schema in a snapshot if the schema history topic is not available.
```

File: documentation/modules/ROOT/pages/connectors/mysql.adoc [hunk 8]
```diff
@@ -3511,10 +3516,44 @@ This option is under consideration for future deprecation, in favor of the `no_d
...
+endif::community[]
+
+ifdef::community[]
+|[[mysql-property-snapshot-mode-configuration-based-snapshot-on-data-error]]<<mysql-property-configuration-based-snapshot-on-data-error, `+snapshot.mode.configuration.based.snapshot.on.data.error+`>>
+|false
+|When `snapshot.mode` is set as `configuration_based`, this setting permits to specify whenever the data should be snapshotted or not in case data is no more present on db log.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/mysql.adoc:3554]:
```suggestion
|If the `snapshot.mode` is set to `configuration_based`, set this property to specify whether the connector includes table data in a snapshot in the event that data is no longer available in the transaction log.
```

File: documentation/modules/ROOT/pages/connectors/oracle.adoc [hunk 9]
```diff
@@ -3159,12 +3164,46 @@ After the snapshot is complete, the connector continues to read change events fr
...
 ifdef::community[]
 `custom`:: The connector performs a snapshot according to the implementation specified by the xref:postgresql-property-snapshot-mode-custom-name[`snapshot.mode.custom.name`] property, which defines a custom implementation of the `io.debezium.spi.snapshot.Snapshotter` interface.
 endif::community[]
 
 For more information, see the xref:oracle-connector-snapshot-mode-options[table of `snapshot.mode` options].
 
+ifdef::community[]
+|[[oracle-property-snapshot-mode-configuration-based-snapshot-data]]<<oracle-property-configuration-based-snapshot-data, `+snapshot.mode.configuration.based.snapshot.data+`>>
+|false
+|When `snapshot.mode` is set as `configuration_based`, this setting permits to specify whenever the data should be snapshotted or not.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/oracle.adoc:3180]:
```suggestion
|If the `snapshot.mode` is set to `configuration_based`, set this property to specify whether the connector includes table data when it performs a snapshot.
```

File: documentation/modules/ROOT/pages/connectors/postgresql.adoc [hunk 10]
```diff
@@ -3304,12 +3309,46 @@ Use this snapshot mode only when you know all data of interest is still reflecte
...
 ifdef::community[]
 `custom`:: The connector performs a snapshot according to the implementation specified by the xref:postgresql-property-snapshot-mode-custom-name[`snapshot.mode.custom.name`] property, which defines a custom implementation of the `io.debezium.spi.snapshot.Snapshotter` interface.
 endif::community[]
 
 For more information, see the xref:#postgresql-connector-snapshot-mode-options[table of `snapshot.mode` options].
 
+ifdef::community[]
+|[[postgresql-property-snapshot-mode-configuration-based-snapshot-data]]<<postgresql-property-configuration-based-snapshot-data, `+snapshot.mode.configuration.based.snapshot.data+`>>
+|false
+|When `snapshot.mode` is set as `configuration_based`, this setting permits to specify whenever the data should be snapshotted or not.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/postgresql.adoc:3325]:
```suggestion
|If the `snapshot.mode` is set to `configuration_based`, set this property to specify whether the connector includes table data when it performs a snapshot.
```

File: documentation/modules/ROOT/pages/connectors/sqlserver.adoc [hunk 11]
```diff
@@ -2872,13 +2877,47 @@ WARNING: Do not use this mode to perform a snapshot if schema changes were commi
...
 `custom`:: The `custom` snapshot mode lets you inject your own implementation of the `io.debezium.spi.snapshot.Snapshotter` interface.
 Set the `snapshot.mode.custom.name` configuration property to the name provided by the `name()` method of your implementation.
 
 For more information, see xref:connector-custom-snapshot[custom snapshotter SPI].
 endif::community[]
 
+ifdef::community[]
+|[[sqlserver-property-snapshot-mode-configuration-based-snapshot-data]]<<sqlserver-property-configuration-based-snapshot-data, `+snapshot.mode.configuration.based.snapshot.data+`>>
+|false
+|When `snapshot.mode` is set as `configuration_based`, this setting permits to specify whenever the data should be snapshotted or not.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/sqlserver.adoc:2894]:
```suggestion
|If the `snapshot.mode` is set to `configuration_based`, set this property to specify whether the connector includes table data when it performs a snapshot.
```

File: documentation/modules/ROOT/pages/connectors/oracle.adoc [hunk 12]
```diff
@@ -3159,12 +3164,46 @@ After the snapshot is complete, the connector continues to read change events fr
...
+endif::community[]
+
+ifdef::community[]
+|[[oracle-property-snapshot-mode-configuration-based-snapshot-schema]]<<oracle-property-configuration-based-snapshot-schema, `+snapshot.mode.configuration.based.snapshot.schema+`>>
+|false
+|When `snapshot.mode` is set as `configuration_based`, this setting permits to specify whenever the schema should be snapshotted or not.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/oracle.adoc:3186]:
```suggestion
|If the `snapshot.mode` is set to `configuration_based`, set this property to specify whether the connector includes the table schema when it performs a snapshot.
```

File: documentation/modules/ROOT/pages/connectors/postgresql.adoc [hunk 13]
```diff
@@ -3304,12 +3309,46 @@ Use this snapshot mode only when you know all data of interest is still reflecte
...
+endif::community[]
+
+ifdef::community[]
+|[[postgresql-property-snapshot-mode-configuration-based-snapshot-schema]]<<postgresql-property-configuration-based-snapshot-schema, `+snapshot.mode.configuration.based.snapshot.schema+`>>
+|false
+|When `snapshot.mode` is set as `configuration_based`, this setting permits to specify whenever the schema should be snapshotted or not.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/postgresql.adoc:3331]:
```suggestion
|If the `snapshot.mode` is set to `configuration_based`, set this property to specify whether the connector includes the table schema when it performs a snapshot.
```

File: documentation/modules/ROOT/pages/connectors/sqlserver.adoc [hunk 14]
```diff
@@ -2872,13 +2877,47 @@ WARNING: Do not use this mode to perform a snapshot if schema changes were commi
...
+endif::community[]
+
+ifdef::community[]
+|[[sqlserver-property-snapshot-mode-configuration-based-snapshot-schema]]<<sqlserver-property-configuration-based-snapshot-schema, `+snapshot.mode.configuration.based.snapshot.schema+`>>
+|false
+|When `snapshot.mode` is set as `configuration_based`, this setting permits to specify whenever the schema should be snapshotted or not.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/sqlserver.adoc:2900]:
```suggestion
|If the `snapshot.mode` is set to `configuration_based`, set this property to specify whether the connector includes the table schema when it performs a snapshot.
```

File: documentation/modules/ROOT/pages/connectors/sqlserver.adoc [hunk 15]
```diff
@@ -2872,13 +2877,47 @@ WARNING: Do not use this mode to perform a snapshot if schema changes were commi
...
+endif::community[]
+
+ifdef::community[]
+|[[sqlserver-property-snapshot-mode-configuration-based-start-stream]]<<sqlserver-property-configuration-based-start-stream, `+snapshot.mode.configuration.based.start.stream+`>>
+|false
+|When `snapshot.mode` is set as `configuration_based`, this setting permits to specify whenever the stream should start or not after snapshot.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/sqlserver.adoc:2906]:
```suggestion
|If the `snapshot.mode` is set to `configuration_based`, set this property to specify whether the connector begins to stream change events after a snapshot completes.
```

File: documentation/modules/ROOT/pages/connectors/oracle.adoc [hunk 16]
```diff
@@ -3159,12 +3164,46 @@ After the snapshot is complete, the connector continues to read change events fr
...
+endif::community[]
+
+ifdef::community[]
+|[[oracle-property-snapshot-mode-configuration-based-start-stream]]<<oracle-property-configuration-based-start-stream, `+snapshot.mode.configuration.based.start.stream+`>>
+|false
+|When `snapshot.mode` is set as `configuration_based`, this setting permits to specify whenever the stream should start or not after snapshot.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/oracle.adoc:3192]:
```suggestion
|If the `snapshot.mode` is set to `configuration_based`, set this property to specify whether the connector begins to stream change events after a snapshot completes.
```

File: documentation/modules/ROOT/pages/connectors/postgresql.adoc [hunk 17]
```diff
@@ -3304,12 +3309,46 @@ Use this snapshot mode only when you know all data of interest is still reflecte
...
+endif::community[]
+
+ifdef::community[]
+|[[postgresql-property-snapshot-mode-configuration-based-start-stream]]<<postgresql-property-configuration-based-start-stream, `+snapshot.mode.configuration.based.start.stream+`>>
+|false
+|When `snapshot.mode` is set as `configuration_based`, this setting permits to specify whenever the stream should start or not after snapshot.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/postgresql.adoc:3337]:
```suggestion
|If the `snapshot.mode` is set to `configuration_based`, set this property to specify whether the connector begins to stream change events after a snapshot completes.
```

File: documentation/modules/ROOT/pages/connectors/oracle.adoc [hunk 18]
```diff
@@ -3159,12 +3164,46 @@ After the snapshot is complete, the connector continues to read change events fr
...
+endif::community[]
+
+ifdef::community[]
+|[[oracle-property-snapshot-mode-configuration-based-snapshot-on-schema-error]]<<oracle-property-configuration-based-snapshot-on-schema-error, `+snapshot.mode.configuration.based.snapshot.on.schema.error+`>>
+|false
+|When `snapshot.mode` is set as `configuration_based`, this setting permits to specify whenever the schema should be snapshotted or not in case schema history topic is missing.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/oracle.adoc:3198]:
```suggestion
|If the `snapshot.mode` is set to `configuration_based`, set this property to specify whether the connector includes table schema in a snapshot if the schema history topic is not available.
```

File: documentation/modules/ROOT/pages/connectors/postgresql.adoc [hunk 19]
```diff
@@ -3304,12 +3309,46 @@ Use this snapshot mode only when you know all data of interest is still reflecte
...
+endif::community[]
+
+ifdef::community[]
+|[[postgresql-property-snapshot-mode-configuration-based-snapshot-on-schema-error]]<<postgresql-property-configuration-based-snapshot-on-schema-error, `+snapshot.mode.configuration.based.snapshot.on.schema.error+`>>
+|false
+|When `snapshot.mode` is set as `configuration_based`, this setting permits to specify whenever the schema should be snapshotted or not in case schema history topic is missing.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/postgresql.adoc:3343]:
```suggestion
|If the `snapshot.mode` is set to `configuration_based`, set this property to specify whether the connector includes table schema in a snapshot if the schema history topic is not available.
```

File: documentation/modules/ROOT/pages/connectors/sqlserver.adoc [hunk 20]
```diff
@@ -2872,13 +2877,47 @@ WARNING: Do not use this mode to perform a snapshot if schema changes were commi
...
+endif::community[]
+
+ifdef::community[]
+|[[sqlserver-property-snapshot-mode-configuration-based-snapshot-on-schema-error]]<<sqlserver-property-configuration-based-snapshot-on-schema-error, `+snapshot.mode.configuration.based.snapshot.on.schema.error+`>>
+|false
+|When `snapshot.mode` is set as `configuration_based`, this setting permits to specify whenever the schema should be snapshotted or not in case schema history topic is missing.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/sqlserver.adoc:2912]:
```suggestion
|If the `snapshot.mode` is set to `configuration_based`, set this property to specify whether the connector includes table schema in a snapshot if the schema history topic is not available.
```

File: documentation/modules/ROOT/pages/connectors/sqlserver.adoc [hunk 21]
```diff
@@ -2872,13 +2877,47 @@ WARNING: Do not use this mode to perform a snapshot if schema changes were commi
...
+endif::community[]
+
+ifdef::community[]
+|[[sqlserver-property-snapshot-mode-configuration-based-snapshot-on-data-error]]<<sqlserver-property-configuration-based-snapshot-on-data-error, `+snapshot.mode.configuration.based.snapshot.on.data.error+`>>
+|false
+|When `snapshot.mode` is set as `configuration_based`, this setting permits to specify whenever the data should be snapshotted or not in case data is no more present on db log.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/sqlserver.adoc:2918]:
```suggestion
|If the `snapshot.mode` is set to `configuration_based`, set this property to specify whether the connector includes table data in a snapshot in the event that data is no longer available in the transaction log.
```

File: documentation/modules/ROOT/pages/connectors/oracle.adoc [hunk 22]
```diff
@@ -3159,12 +3164,46 @@ After the snapshot is complete, the connector continues to read change events fr
...
+endif::community[]
+
+ifdef::community[]
+|[[oracle-property-snapshot-mode-configuration-based-snapshot-on-data-error]]<<oracle-property-configuration-based-snapshot-on-data-error, `+snapshot.mode.configuration.based.snapshot.on.data.error+`>>
+|false
+|When `snapshot.mode` is set as `configuration_based`, this setting permits to specify whenever the data should be snapshotted or not in case data is no more present on db log.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/oracle.adoc:3204]:
```suggestion
|If the `snapshot.mode` is set to `configuration_based`, set this property to specify whether the connector includes table data in a snapshot in the event that data is no longer available in the transaction log.
```

File: documentation/modules/ROOT/pages/connectors/postgresql.adoc [hunk 23]
```diff
@@ -3304,12 +3309,46 @@ Use this snapshot mode only when you know all data of interest is still reflecte
...
+endif::community[]
+
+ifdef::community[]
+|[[postgresql-property-snapshot-mode-configuration-based-snapshot-on-data-error]]<<postgresql-property-configuration-based-snapshot-on-data-error, `+snapshot.mode.configuration.based.snapshot.on.data.error+`>>
+|false
+|When `snapshot.mode` is set as `configuration_based`, this setting permits to specify whenever the data should be snapshotted or not in case data is no more present on db log.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/postgresql.adoc:3349]:
```suggestion
|If the `snapshot.mode` is set to `configuration_based`, set this property to specify whether the connector includes table data in a snapshot in the event that data is no longer available in the transaction log.
```

**Review by roldanbob — CHANGES_REQUESTED** [review]
Proposed suggestions and one set of questions wrt the use case for `snapshot.mode.configuration.based.snapshot.on.data.error`.

File: documentation/modules/ROOT/pages/connectors/db2.adoc [hunk 24]
```diff
@@ -2710,13 +2715,48 @@ WARNING: Do not use this mode to perform a snapshot if schema changes were commi
...
+ifdef::community[]
+|[[db2-property-snapshot-mode-configuration-based-snapshot-on-schema-error]]<<db2-property-configuration-based-snapshot-on-schema-error, `+snapshot.mode.configuration.based.snapshot.on.schema.error+`>>
+|false
+|When `snapshot.mode` is set as `configuration_based`, this setting permits to specify whenever the schema should be snapshotted or not in case schema history topic is missing.
+endif::community[]
+
+ifdef::community[]
+|[[db2-property-snapshot-mode-configuration-based-snapshot-on-data-error]]<<db2-property-configuration-based-snapshot-on-data-error, `+snapshot.mode.configuration.based.snapshot.on.data.error+`>>
+|false
+|When `snapshot.mode` is set as `configuration_based`, this setting permits to specify whenever the data should be snapshotted or not in case data is no more present on db log.
```
[the remark this answers is not in the document]
Comment (mfvitale) [reply documentation/modules/ROOT/pages/connectors/db2.adoc:2757]: This regard the last commited offset. If it is not available on the log (ie. purged), set this property to `true` permits to re do a snapshot.

[the remark this answers is not in the document]
Comment (roldanbob) [reply documentation/modules/ROOT/pages/connectors/db2.adoc:2757]: Ok, thanks for the clarification. I've made some changes. WDYT?

[the remark this answers is not in the document]
Comment (mfvitale) [reply documentation/modules/ROOT/pages/connectors/db2.adoc:2757]: @roldanbob Sounds good!

**Review by roldanbob — APPROVED** [review]
LGTM

**Pull request merged**